docs/spec/v2alpha1/helmreleases.md

```diff
@@ -0,0 +1,276 @@
+
+	// Suspend tells the reconciler to suspend reconciliations for
+	// this Helm release. Defaults to false.
+	// +optional
+	Suspend bool `json:"suspend,omitempty"`
+
+	// Timeout for Helm actions performed. Defaults to 'Interval' duration.
+	// +optional
+	Timeout *metav1.Duration `json:"timeout,omitempty"`
+
+	// Wait tells the reconciler to wait with marking a Helm action as
+	// successful until all resources are in a ready state.
+	// If set, it will wait for as long as 'Timeout'.
+	// +optional
+	Wait bool `json:"wait,omitempty"`
+
+	// MaxHistory is the number of revisions saved by Helm for this release.
+	// Use '0' for an unlimited number of revisions; defaults to '10'.
+	// +optional
+	MaxHistory *int `json:"maxHistory,omitempty"`
+
+	// Install holds the configuration for performing Helm install actions.
+	// +optional
+	Install *struct {
+		// Timeout for the Helm install action.
+		// Defaults to the main 'Timeout' duration.
+		// +optional
+		Timeout *metav1.Duration `json:"timeout,omitempty"`
+
+		// Wait tells the reconciler to wait with marking the Helm install as
+		// successful until all resources are in a ready state. When set, it will
+		// wait for as long as 'Timeout'. Defaults to the main 'Wait' value.
+		// +optional
+		Wait bool `json:"wait,omitempty"`
+
+		// SkipCRDs will mark this Helm release to skip the creation of CRDs during
+		// the installation.
+		// +optional
+		SkipCRDs bool `json:"skipCRDs,omitempty"`
+	} `json:"install,omitempty"`
+	
+	// Upgrade holds the configuration for performing Helm upgrade actions.
+	// +optional
+	Upgrade *struct {
+		// Timeout for the Helm upgrade action. Defaults to the main
+		// 'Timeout' duration.
+		// +optional
+		Timeout *metav1.Duration `json:"timeout,omitempty"`
+
+		// Wait tells the reconciler to wait with marking the Helm upgrade as
+		// successful until all resources are in a ready state. When set, it will
+		// wait for as long as 'Timeout'. Defaults to the main 'Wait' value.
+		// +optional
+		Wait bool `json:"wait,omitempty"`
+
+		// ResetValues resets all values to the chart defaults, so that the
+		// configuration applied is what is defined in the resource values.
+		// Defaults to true to ensure declarative behaviour.
+		// +optional
+		ResetValues *bool `json:"resetValues,omitempty"`
+	} `json:"upgrade,omitempty"`
+
+	// Test holds the configuration for enabling and performing Helm
+	// test actions.
+	// +optional
+	Test *struct {
+		// Enable tells the reconciler to perform a test action
+		// on the configured condition reasons. Defaults to false.
+		// +optional
+		Enable bool `json:"enable,omitempty"`
+
+		// OnConditionReason defines on what condition reasons a Helm test
+		// should be performed. Defaults to ['InstallSucceeded', 'UpgradeSucceeded'].
+		// +optional
+		OnConditionReason []string `json:"onConditionReason,omitempty"`
+
+		// Timeout for the Helm test action. Defaults to the main
+		// 'Timeout' duration.
```

Comment:
I think an explicit default, matching that of the `helm test` command (300s) which is what is currently implemented in the Helm Operator is more intuitive. I don't think the timeout for the release itself it closely related to the timeout for its tests.